nmmo/core/realm.py

```diff
@@ -72,6 +72,10 @@ def reset(self, map_id: int = None):
         idx: Map index to load
     """
     self.log_helper.reset()
+    # reset datastore tables, except the table for TileState
+    for s in [EntityState, ItemState]:
```

Comment:
What do you think about doing:

EntityState.clear_datastore()
ItemState.clear_datastore()

This would avoid protected-access calls.

Also, we could put this into players.reset() but I'm fine with it here

Reply:
The relationship between SerializedState (like EntityState, ItemState), DataStore, and DataTable was hard to understand for me. EntityState and ItemState don't seem to hold actual data, and I needed to provide `object_type: str`, which is `_name` of SerializedState, to DataStore to get access to the numpy tables.

Is it necessary to keep the name of SerializedState private?

Reply:
I see that players.reset() does delete player entries in the numpy tables. But I also saw that the id allocators were not reset, so the deterministic test failed (the ids were not the same after resetting the env).

Reply:
clear_datastore() would be a new method you would define on the State object in the subclass() method.

I think it might actually be better to reset the datastore in the determinism test. There isn't any reason to reset the datastore in realm, maybe we can just assert that the datastore is empty during reset().